scikeras/utils/transformers.py

```diff
@@ -0,0 +1,194 @@
+
+    def inverse_transform(
+        self, X: np.ndarray, return_proba: bool = False
+    ) -> np.ndarray:
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        if self._target_type == "binary":
+            # array([0.9, 0.1], [.2, .8]) -> array(['yes', 'no'])
+            if y.ndim == 1 or (y.shape[1] == 1 and self.n_classes_ == 2):
+                # result from a single sigmoid output
+                # reformat so that we have 2 columns
+                y = np.column_stack([1 - y, y])
+            y_ = np.argmax(y, axis=1).reshape(-1, 1)
+            class_predictions = self.final_encoder_.inverse_transform(y_)
+        elif self._target_type == "multiclass":
+            # array([0.8, 0.1, 0.1], [.1, .8, .1]) ->
+            # array(['apple', 'orange'])
+            idx = np.argmax(y, axis=-1)
+            if not is_categorical_crossentropy(self.loss):
+                y_ = idx.reshape(-1, 1)
+            else:
+                y_ = np.zeros(y.shape, dtype=int)
+                y_[np.arange(y.shape[0]), idx] = 1
+            class_predictions = self.final_encoder_.inverse_transform(y_)
+        elif self._target_type == "multiclass-onehot":
+            # array([.8, .1, .1], [.1, .8, .1]) ->
+            # array([[1, 0, 0], [0, 1, 0]])
+            idx = np.argmax(y, axis=-1)
+            y_ = np.zeros(y.shape, dtype=int)
+            y_[np.arange(y.shape[0]), idx] = 1
+            class_predictions = y_
+        elif self._target_type == "multilabel-indicator":
+            class_predictions = np.around(y)
+        else:
+            raise NotImplementedError(
+                "Class-predictions are not clearly defined for"
+                " 'multiclass-multioutput' target types."
+            )
+
+        if return_proba:
+            return y
+        return np.squeeze(np.column_stack(class_predictions)).astype(
+            self.y_dtype_, copy=False
+        )
+
+    def get_metadata(self):
+        return {
+            "classes_": self.classes_,
+            "n_classes_": self.n_classes_,
+            "n_outputs_": self.n_outputs_,
+            "model_n_outputs_": self.model_n_outputs_,
+        }
+
+
+class RegressorTargetEncoder(BaseEstimator, TransformerMixin):
+    """Default target transformer for KerasRegressor.
+    """
+
+    def fit(self, X: np.ndarray) -> "RegressorTargetEncoder":
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        self.y_dtype_ = y.dtype
+        self.n_outputs_ = 1 if y.ndim == 1 else y.shape[1]
+        self.model_n_outputs_ = 1
+        return self
+
+    def transform(self, X: np.ndarray) -> np.ndarray:
+        y = X
+        n_outputs_ = 1 if y.ndim == 1 else y.shape[1]
+        if n_outputs_ != self.n_outputs_:
+            raise ValueError(
+                f"Detected `y` to have {n_outputs_},"
+                f" but this {self} expects"
+                f" {self.n_outputs_} for `y`."
```

Comment:
Style nit: I don't know if `{self}` is necessary (I've seen this other places too). I'd say "this class" instead.

Reply:
It should be `{self.__class__.__name__}`. I'll fix it. Thanks.

Reply:
I only found that in this one place. As for why the customization in the first place, it comes from sklearn: https://github.com/scikit-learn/scikit-learn/blob/e6555decf8a72958d26d499deb17aabca41a562a/sklearn/base.py#L432

I guess it may be useful to know which model caused an error in a pipeline, ensemble, etc?

Reply:
> I guess it may be useful to know which model caused an error in a pipeline, ensemble, etc?

Oh, I see. I support this now.